### PR TITLE
feat: Add onrevert to zevm connector

### PR DESCRIPTION
**Pull request by andresaiello**
### Summary
- Add onrevert and extend onReceive
- Deployed on Athens3 on 0xd2aA152341E19eEFf4baa2C704B0B56523a3812f

### CODE REVIEW

File: contracts/zevm/ZetaConnectorZEVM.sol [hunk 1]
```diff
@@ -61,38 +13,36 @@ contract ZetaConnectorZEVM is ZetaInterfaces {
...
+     */
+    modifier onlyFungibleModule() {
+        if (msg.sender != FUNGIBLE_MODULE_ADDRESS) revert OnlyFungibleModule();
+        _;
     }
 
+    constructor(
+        address zetaTokenAddress_,
+        address tssAddress_,
+        address tssAddressUpdater_,
```
Comment (lumtis) [inline contracts/zevm/ZetaConnectorZEVM.sol:28]: There is no notion of TSS for ZetaChain. Observers don't interact with ZetaChain with the TSS, so keeping this notion in zevm smart contract might lead to confusion.

File: contracts/zevm/ZetaConnectorZEVM.sol [hunk 2]
```diff
@@ -61,38 +13,36 @@ contract ZetaConnectorZEVM is ZetaInterfaces {
...
+        address pauserAddress_
+    ) ZetaConnectorBase(zetaTokenAddress_, tssAddress_, tssAddressUpdater_, pauserAddress_) {}
```
Comment (lumtis) [inline contracts/zevm/ZetaConnectorZEVM.sol:30]: What is the purpose of using `ZetaConnectorBase` if the logic for the zevm version is different (using fungible address for permission)?

Comment (lumtis) [inline contracts/zevm/ZetaConnectorZEVM.sol:30]: Also the current plan is to overwrite the bytecode of the connector zevm on live network so the ideal would be to not change the state layout of the smart contract

**Review by lumtis — COMMENTED** [review]
I think we should add tests for the new functions in the same PR

File: contracts/zevm/ZetaConnectorZEVM.sol (hunk 2 above)
Comment (andresaiello) [reply contracts/zevm/ZetaConnectorZEVM.sol:30]: the idea was to guarantee have the same interfaces but I will copy the ones we need

**Review by lumtis — APPROVED** [review]
LGTM

**Review by kingpinXD — APPROVED** [review]

**Pull request merged**